src/output.c

```diff
@@ -55,18 +55,40 @@ const struct output output_wl = {
 };
 #endif
 
+const struct output* get_x11_output() {
+        const struct output* output = &output_x11;
+        if (output->init()) {
+                return output;
+        } else {
+                LOG_E("Couldn't initialize X11 output. Aborting...");
+        }
+}
+
+#ifdef ENABLE_WAYLAND
+const struct output* get_wl_output() {
+        const struct output* output = &output_wl;
+        if (output->init()) {
+                return output;
+        } else {
+                LOG_W("Couldn't initialize wayland output. Falling back to X11 output.");
+                output->deinit();
```

Comment:
`deinit` is a noop for wayland. Perhaps we should properly deinit to avoid any interactions with having a partially initialized backend.

Reply:
That was my intention, but the deinit was linked to the wrong function. I linked it now, and fixed the wayland deinit. I'm not very confident everything is deallocated, so take a close look at it. It doesn't crash at least when exiting from every possible abort point.

Reply:
Deallocation isn't that important, it's only a few bytes. The main point is to make sure that no events fire on the wayland side. This would confuse the main logic and be a debugging hell :)